docs/source/guides/data.rst

```diff
@@ -0,0 +1,548 @@
+    mnist_train = DataLoader(mnist_train, batch_size=64)
+
+.. testoutput::
+    :hide:
+    :skipif: os.path.isdir(os.path.join(os.getcwd(), 'MNIST')) or not _TORCHVISION_AVAILABLE
+
+    Downloading ...
+    Extracting ...
+    Downloading ...
+    Extracting ...
+    Downloading ...
+    Extracting ...
+    Processing...
+    Done!
+
+Using DataLoaders in Lightning
+==============================
+
+You can pass DataLoaders to the Lightning Trainer in 3 ways:
+
+1. Pass DataLoaders to Trainer.fit()
+------------------------------------
+Pass in the dataloaders to the :meth:`pytorch_lightning.trainer.trainer.Trainer.fit` function.
+
+.. code-block:: python
+
+    model = LitMNIST()
+    trainer = Trainer()
+    trainer.fit(model, mnist_train)
+
+
+2. Pass DataLoaders to the LightningModule
+------------------------------------------
+For fast research prototyping, it might be easier to add the DataLoaders to your :class:`~pytorch_lightning.core.lightning.LightningModule`, using the DataLoader hooks (:meth:`~pytorch_lightning.core.lightning.LightningModule.train_dataloader`, :meth:`~pytorch_lightning.core.lightning.LightningModule.val_dataloader`, :meth:`~pytorch_lightning.core.lightning.LightningModule.test_dataloader`).
+
+
+.. code-block:: python
+
+    class LitMNIST(pl.LightningModule):
+
+        def train_dataloader(self):
+            transforms = ...
+            mnist_train = ...
+            return DataLoader(mnist_train, batch_size=64)
+
+        def val_dataloader(self):
+            transforms = ...
+            mnist_val = ...
+            return DataLoader(mnist_val, batch_size=64)
+
+        def test_dataloader(self):
+            transforms = ...
+            mnist_test = ...
+            return DataLoader(mnist_test, batch_size=64)
+
+The :class:`~pytorch_lightning.core.lightning.LightningModule` contains the DataLoaders, so there's no need to specify on :meth:`~pytorch_lightning.trainer.trainer.Trainer.fit`.
+
+.. code-block:: python
+
+    model = LitMNIST()
+    trainer = Trainer()
+    trainer.fit(model)
+
+3. LightningDataModules (recommended)
+-------------------------------------
+Defining free-floating dataloaders, splits, download instructions, and such can get messy.
+We recommend grouping the full definition of the datasets into a :class:`~pytorch_lightning.core.datamodule.LightningDataModule` which includes:
+
+- Download instructions
+- Processing instructions
+- Split instructions
+- Train dataloader
+- Val dataloader(s)
+- Test dataloader(s)
+
+.. testcode::
+
+    class MyDataModule(LightningDataModule):
+
+        def __init__(self):
+            super().__init__()
+            self.train_dims = None
+            self.vocab_size = 0
+
+        def prepare_data(self):
+            # called only on 1 GPU
+            download_dataset()
+            tokenize()
+            build_vocab()
+
+        def setup(self, stage: Optional[str] = None):
+            # called on every GPU
```

Comment:
```suggestion
            # called on every Process
```

I think for DP and DDP2 it is only called once since the process manages multiple GPUs